icu4c/source/i18n/measunit.cpp

```diff
@@ -2271,9 +2237,11 @@ StringEnumeration* MeasureUnit::getAvailableTypes(UErrorCode &errorCode) {
 }
 
 bool MeasureUnit::findBySubType(StringPiece subType, MeasureUnit* output) {
+    // Sanity checking kCurrencyOffset and final entry in gOffsets
+    U_ASSERT(uprv_strcmp(gTypes[kCurrencyOffset], "currency") == 0);
+    U_ASSERT(gOffsets[UPRV_LENGTHOF(gOffsets) - 1] == UPRV_LENGTHOF(gSubTypes));
+
     for (int32_t t = 0; t < UPRV_LENGTHOF(gOffsets) - 1; t++) {
-        // Ensure kCurrencyOffset is set correctly
-        U_ASSERT(uprv_strcmp(gTypes[kCurrencyOffset], "currency") == 0);
```

Comment:
I guess it was from pure inattentiveness that I placed this assert inside the for loop before.
I feel this assert is partially documentation for how the "skip currency units" code works, so it fits better for me than the assert I'm adding in this PR:

It doesn't feel like this function is the ideal place for the new assert, but I don't find a "more ideal" place. (In Java, a "static intialization block" might have served?)